sim/core/sim.go

```diff
@@ -450,6 +472,14 @@ func (sim *Simulation) Step() bool {
 	last := len(sim.pendingActions) - 1
 	pa := sim.pendingActions[last]
 
+	if pa.NextActionAt >= sim.minWeaponAttackTime && sim.minWeaponAttackTime <= sim.minTaskTime {
```

Comment:
It seems like we now have this idea of 'minimum timestamp amongst a group of events' in a few places (at least here and in aura trackers). Perhaps there would be a way to write a generic utility to manage it? Would definitely be in a separate PR.

Reply:
The "Task" interface and related methods was basically a shot at such a generic utility, but it's not quite there yet:
- it doesn't allow grouping tasks, which is important since grouping allows to minimize the costs of finding a "global" minimum
- it doesn't allow to inline the "ready" checks
Both problems can be tackled, of course.